src/Core/Checkout/Cart/Event/AdminPromotionCodeRedeemedEvent.php

```diff
@@ -0,0 +1,35 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Cart\Event;
+
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Contracts\EventDispatcher\Event;
+
+#[Package('checkout')]
+class AdminPromotionCodeRedeemedEvent extends Event
```

Comment:
I personally don't like the admin prefix, since this all can also be done via API.
Maybe `RecalculationPromotionLineItemAddedEvent`?

Reply:
`RecalculationPromotionIndividualCodeAddedEvent`? Suits the purpose of the event better.

Reply:
I don't like it, because it is not triggered only if individual codes are added, but if any code is added, right? The event should not be named based on the Subscriber